ledger/artifactmanager/artifactmanager.go

```diff
@@ -35,8 +35,8 @@ func (m *LedgerArtifactManager) checkRequestRecord(requestID record.ID) error {
 	return nil
 }
 
-func (m *LedgerArtifactManager) getCodeRecord(codeID record.ID) (*record.CodeRecord, error) {
-	rec, err := m.storer.GetRecord(codeID)
+func (m *LedgerArtifactManager) getCodeRecord(codeRef record.Reference) (*record.CodeRecord, error) {
```

Comment:
Maybe we should pass codeRef by reference here as well?
Same for other methods like getCodeRecordCode